### PR TITLE
share blog button

### LINKED ISSUE #684: Blog: Copy feature (share)
### What happened?

Blog posts can have a "copy blog post url", similar to the copy function in the events page.
image

Another suggestion is to have a "copy entire blog as plaintext".

### PR DESCRIPTION
**Pull request by caretcakejake**
Resolves #684 
👽🤟

### CODE REVIEW

File: src/lib/components/blog/blog-post-preview.svelte [hunk 1]
```diff
@@ -24,6 +26,20 @@
           {/if}</a
         >
       </div>
+      <button
+        class="clipboard-btn"
+        on:click={(ev) => {
+          ev.preventDefault();
```
Comment (jaasonw) [inline src/lib/components/blog/blog-post-preview.svelte]: prefer using svelte event modifier
image

File: src/lib/components/blog/blog-post-preview.svelte [hunk 2]
```diff
@@ -97,4 +113,13 @@
...
     gap: 1em;
   }
+
+  .clipboard-btn {
+    display: flex;
+    align-items: center;
+    margin-left: auto;
+    border: none;
+    background-color: transparent;
+    color: var(--text-color);
```
Comment (jaasonw) [inline src/lib/components/blog/blog-post-preview.svelte]: button should prob have
a. `cursor: pointer;` on hover
b. a border similar to the event page
c. a hover effect similar to that of the event page

https://github.com/EthanThatOneKid/acmcsuf.com/assets/8981287/30def6d7-1941-412f-aef7-8e9f961dedc3

File: src/lib/components/blog/blog-post-preview.svelte [hunk 3]
```diff
@@ -24,6 +26,20 @@
           {/if}</a
         >
       </div>
+      <button
+        class="clipboard-btn"
+        on:click={(ev) => {
+          ev.preventDefault();
+          copy(
+            "https://acmcsuf.com/" + makeBlogPostPageUrl(post.id),
```
Comment (jaasonw) [inline src/lib/components/blog/blog-post-preview.svelte]: avoid hard coding the url here, the current url can be accessed with `window.location.href` or `document.URL`
image

File: src/lib/components/blog/blog-post-preview.svelte [hunk 4]
```diff
@@ -24,6 +26,20 @@
...
+            'Copied post link to clipboard!',
+            'Failed to copy post link to clipboard!',
+            'general'
```
Comment (jaasonw) [inline src/lib/components/blog/blog-post-preview.svelte]: this parameter is optional and doesn't really need to be here

**Review by jaasonw — CHANGES_REQUESTED** [review]

File: src/lib/components/blog/blog-post-preview.svelte (hunk 4 above)
Comment (caretcakejake) [reply src/lib/components/blog/blog-post-preview.svelte]: you mean line 36?

Comment (jaasonw) [reply src/lib/components/blog/blog-post-preview.svelte]: i mean line 37, "general"

File: src/lib/components/blog/blog-post-preview.svelte (hunk 2 above)
Comment (caretcakejake) [reply src/lib/components/blog/blog-post-preview.svelte]: LMAOO not the snapchat record 💀 
>  Snapchat-1326091595.mp4

File: src/lib/components/blog/blog-post-preview.svelte (hunk 4 above)
Comment (caretcakejake) [reply src/lib/components/blog/blog-post-preview.svelte]: This makes the toast popup blue which makes sense and Karni liked it

Comment (jaasonw) [reply src/lib/components/blog/blog-post-preview.svelte]: it was blue when i tested it without but I can double check

File: src/lib/components/blog/blog-post-preview.svelte (hunk 2 above)
Comment (caretcakejake) [reply src/lib/components/blog/blog-post-preview.svelte]: No the button looks fine. No highlights 😠

File: src/lib/components/blog/blog-post-preview.svelte (hunk 3 above)
Comment (caretcakejake) [reply src/lib/components/blog/blog-post-preview.svelte]: is done

**Review by jaasonw — APPROVED** [review]
yes

**Comment by EthanThatOneKid** [comment]
@barrotbake The reason the check failed for this PR is because of a formatting error. This can be resolved by committing the changes made after running `npm run format`. Can you let me know if you'd like me to go ahead and format and then merge the PR?

**Comment by caretcakejake** [comment]
Hey @EthanThatOneKid yeah you can go ahead and run it and merge. Thanks so much!

**Review by EthanThatOneKid — APPROVED** [review]
LGTM 😎

**Pull request merged**